src/base/projects/BranchSelector.svelte

```diff
@@ -0,0 +1,123 @@
+<script lang="ts">
+  import { getBranchesByCommit, Info } from "@app/project";
+  import { formatCommit } from "@app/utils";
+
+  export let branches: [string, string][];
+  export let project: Info;
+  export let selectedBranch: [string, string] | null;
+  export let commit: string;
+
+  let branchesDropdown = false;
+  const switchBranch = (name: string, head: string) => {
+    commit = head;
+    selectedBranch = [name, head];
+  };
+
+  $: branchNames = getBranchesByCommit(commit, branches);
```

Comment:
Since there could be commits that are head of multiple branches, we return here a stringarray with all the branch names that fit the commit. and show them in the UI as eg. `master|dev`

Reply:
How is this used? I'm not able to try this out? This should only be relevant when viewing a commit, which we don't have right now.

Reply:
Or perhaps it's used when viewing the anchored state, and that happens to belong to be the same across multiple branches? In that case I think only the default branch is relevant.

Reply:
![Screenshot from 2022-01-10 13-49-28](https://user-images.githubusercontent.com/7912302/148768575-2c1b2603-d909-4540-a679-d9ec72cf146b.png)
The issue is when you have two branches pointing to the same commit, shouldn't be something that occurs often, but I think it can be a possible edge case.
In that case which branch name should we display? I went for both :sunglasses:

Reply:
Hmm I think the problem here is that we actually don't support branch names in the route. If a user selects `dev`, it should show `dev`. It doesn't matter that other branches point to the same commit. But because the branch selector just selects commits, we have this problem. Could we have a branch path? eg. `/tree/master`?

Reply:
ok this has been implemented in c1e2717cae73295799262c6a6b02e52d988653c4